File: android/src/main/kotlin/cz/appsdevteam/gptom/models/ResultCodes.kt
package cz.appsdevteam.gptom.models

object ResultCodes {
    const val OK = "ok"
    const val NOT_INITIALIZED = "notInitialized"
    const val NOT_INSTALLED = "notInstalled"
    const val UNSUPPORTED_ON_PLATFORM = "unsupportedOnPlatform"
    const val INVALID_ARGUMENT = "invalidArgument"
    const val TIMEOUT = "timeout"
    const val SERVICE_BIND_FAILED = "serviceBindFailed"
    const val INTERNAL_ERROR = "internalError"
    const val FAILED = "failed"
    const val NETWORK_ERROR = "networkError"
    const val INVALID_CLIENT_ID = "invalidClientId"
    const val MERCHANT_INFO_MISSING = "merchantInfoMissing"
    const val FAILED_TAP_TO_PAY = "failedTapToPay"
    const val FAILED_TO_CLOSE_BATCH = "failedToCloseBatch"
    const val INVALID_DEEPLINK = "invalidDeeplink"
}

File: ios/Classes/models/ResultCodes.swift
enum ResultCodes {
    static let ok = "ok"
    static let notInitialized = "notInitialized"
    static let notInstalled = "notInstalled"
    static let unsupportedOnPlatform = "unsupportedOnPlatform"
    static let invalidArgument = "invalidArgument"
    static let timeout = "timeout"
    static let serviceBindFailed = "serviceBindFailed"
    static let internalError = "internalError"
    static let failed = "failed"
    static let networkError = "networkError"
    static let invalidClientId = "invalidClientId"
    static let merchantInfoMissing = "merchantInfoMissing"
    static let failedTapToPay = "failedTapToPay"
    static let failedToCloseBatch = "failedToCloseBatch"
    static let invalidDeeplink = "invalidDeeplink"

    static func from(_ error: DeeplinkError?) -> String {
        guard let error else {
            return ResultCodes.failed
        }

        switch error {
        case .failed:
            return ResultCodes.failed

        case .networkError:
            return ResultCodes.networkError

        case .invalidClientId:
            return ResultCodes.invalidClientId

        case .merchantInfoMissing:
            return ResultCodes.merchantInfoMissing

        case .failedTapToPay:
            return ResultCodes.failedTapToPay

        case .failedToCloseBatch:
            return ResultCodes.failedToCloseBatch
        }
    }
}
